src/arch/x86/idle.c

```diff
@@ -7,11 +7,21 @@
 #include <config.h>
 #include <api/debug.h>
 
-void idle_thread(void)
+/*
+ * The idle thread does not have a dedicated stack and runs in
+ * the context of the idle thread TCB. Make sure that the compiler
+ * always eliminates the function prologue by declaring the
+ * idle_thread with the naked attribute.
+ */
+__attribute__((naked)) void idle_thread(void)
```

Comment:
should add the noreturn attribute also

Reply:
Yeah true the function should be marked `noreturn`. I think it's better to do that in a separate commit though to keep changes small.

Reply:
just add the commit to this PR then.

Reply:
Done!